src/coordination/coordinator_instance.cpp

```diff
@@ -66,11 +67,28 @@ CoordinatorInstance::CoordinatorInstance(CoordinatorInstanceInitConfig const &co
   raft_state_->InitRaftServer();
 }
 
+CoordinatorInstance::~CoordinatorInstance() {
+  // Order is important:
+  // 1. Thread pool might be running with force reset, or become follower, so we should shut that down
+  // 2. Await shutdown of coordinator thread pool so that coordinator can't become follower or do force reset
+  // 3. Frequent checks running, we need to stop them before raft_state_ is destroyed
+  ShuttingDown();
+  thread_pool_.Shutdown();
```

Comment:
Why this explicit handling of shutdown? Why is this necessary, for what?
Can you write also in code comment about that. CoordinatorInstance has become complex so maybe it wouldn't hurt

Reply:
done